market_analysis/views/default.py

```diff
@@ -0,0 +1,450 @@
+                    Association.user_id == current_user_id))\
+            .update({Association.shares: updated_amount})
+
+    list_of_stock_ids = []
+    for row in query:
+        list_of_stock_ids.append(row.child.symbol)
+    if len(list_of_stock_ids) == 0:
+        return HTTPFound(location=request.route_url('search'))
+    else:
+        elements = []
+        for stock in list_of_stock_ids:
+            elements.append({'Symbol': str(stock),
+                             'Type': 'price', 'Params': ['c']})
+        return build_graph(request, elements, True)
+
+
+def package_data(info, entries, msg):
+    """Package data for single_stock_details."""
+    try:
+        info['info'] = entries
+        info['msg'] = msg
+    except TypeError:
+        info = {}
+        info['info'] = entries
+        msg = 'Trouble connecting to API.'
+        info['msg'] = msg
+    return info
+
+
+def check_bad_msg(entries):
+    """Return True if bad request."""
+    if 'Message' in entries.keys():
+        return True
+    return False
+
+
+@view_config(route_name='details', renderer="../templates/details.jinja2",
+                        permission="secret")
+def single_stock_details(request):
+    """Details for single-stock."""
+    entries = {}
+    elements = []
+    msg = ''
+    sym = request.matchdict['sym']
+    resp = requests.get('http://dev.markitondemand.com/'
+                        'Api/v2/Quote/json?symbol=' + sym)
+    if resp.status_code == 200:
+        entries = {key: value for key, value in resp.json().items()}
+        if check_bad_msg(entries):
+            msg = 'Bad request.'
+        elements.append({'Symbol': str(sym), 'Type': 'price', 'Params': ['c']})
+    else:
+        msg = 'Could not fulfill the request.'
+    temp = build_graph(request, elements)
+    temp = package_data(temp, entries, msg)
+    return temp
+
+
+@view_config(route_name='admin', renderer="../templates/admin.jinja2",
+             permission='admin')
+def admin(request):
+    '''
+    A page to display a users information to the site adimn and allow
+    them to change and update user information, or remove user.
+    '''
+    message = user_to_delete = ''
+    if request.method == 'POST':
+        if (request.POST['username'] != 'DELETE_ME_NOW') and \
+           (request.POST['username'] != 'CANCEL'):
+            username = request.POST['username']
+            message = 'Are you sure you want to delete user {}?'\
+                      .format(username)
+            request.session['user_to_delete'] = username
+        elif request.POST['username'] == 'DELETE_ME_NOW':
+            username = request.session['user_to_delete']
+            query = request.dbsession.query(Users)\
+                .filter(Users.username == username).first()
+            request.dbsession.delete(query)
+            request.session['user_to_delete'] = ''
+        elif request.POST['username'] == 'CANCEL':
+            request.session['user_to_delete'] = ''
```

Comment:
A second thought, it's more common to put this type of "confirm/cancel" action on the client side.  The user clicks "delete" and javascript intercepts the form submission to show a confirmation box.  If 'cancel' is clicked, no request is ever sent to the backend.  If 'confirm' is clicked, then javascript can inject a "confirm" value into the form data to be submitted.  It's a bit more efficient to do things this way since you may not end up needing to make any requests to the server, and if you do need to, you only need to make one.